orderbook/src/main.rs

```diff
@@ -160,34 +160,41 @@ async fn main() {
         .expect("failed to create gas price estimator"),
     );
 
-    let unsupported_tokens = HashSet::from_iter(args.shared.unsupported_tokens);
+    let pair_providers =
+        pool_aggregating::pair_providers(&args.shared.baseline_sources, chain_id, &web3).await;
+
     let mut base_tokens = HashSet::from_iter(args.shared.base_tokens);
     // We should always use the native token as a base token.
     base_tokens.insert(native_token.address());
+    let mut allowed_tokens = args.shared.allowed_tokens;
+    allowed_tokens.extend(base_tokens.iter().copied());
+    let unsupported_tokens = args.shared.unsupported_tokens;
     assert!(
-        unsupported_tokens
-            .intersection(&base_tokens)
-            .collect::<HashSet<_>>()
-            .is_empty(),
-        "Base tokens include at least one unsupported token!"
+        allowed_tokens
+            .iter()
+            .all(|token| !unsupported_tokens.contains(token)),
+        "same token can't be both allowed and denied"
```

Comment:
nit: Maybe this could move into a `ListBasedDetector::new()` and make the fields private, thus making it illegal to create an instance with tokens that are both simultaneously allowed and denied?

Reply:
added